Objects/dictobject.c

```diff
@@ -614,6 +614,61 @@ new_dict_with_shared_keys(PyDictKeysObject *keys)
     return new_dict(keys, values);
 }
 
+
+static PyObject *
+clone_dict(PyObject *s)
+{
+    assert(PyDict_CheckExact(s));
+
+    PyDictKeysObject *keys;
+    PyDictObject *new;
+    PyDictObject *orig = (PyDictObject*)s;
+    Py_ssize_t i, n, keys_size;
+    PyDictKeyEntry *entry, *ep0;
+    PyObject *value;
```

Comment:
I meant all local variables can be defined where it's initialized.
for example, 
```c
for (Py_ssize_t i = 0; i < keys->dk_nentries; i++) {
    PyDictKeyEntry *entry = &ep0[i];
...
```

https://google.github.io/styleguide/cppguide.html#Local_Variables
> C++ allows you to declare variables anywhere in a function. We encourage you to declare them in as local a scope as possible, and as close to the first use as possible. This makes it easier for the reader to find the declaration and see what type the variable is and what it was initialized to. 

C99 is same to C++ in this case.

Reply:
Done